### ISSUE TITLE
Add Check Support for Java 21 Pattern Matching for Switch Syntax: OperatorWrap

### ISSUE THREAD

**Issue by mahfouz72** [checkstyle/checkstyle#14962]
child of #14961 
we should support the new syntax of guarded pattern (`LITERAL_WHEN`) in `OperatorWrap` check. after adding support for `LITERAL_WHEN`
we need to update [this file](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapGuardedPatterns.java) to use the new syntax

**Comment by rnveach** [comment]
https://checkstyle.org/checks/whitespace/operatorwrap.html

Do we have something that says `when` is an operator?

**Comment by mahfouz72** [comment]
JLS: https://docs.oracle.com/javase/specs/jls/se21/html/jls-3.html#jls-3.12
they are listing the mathematical logical bitwise operators so `when` as well as method reference and `instanceof` are not considered operators

 I think we are not being strict here that the check tokens should be formal operators

Does this block us from supporting when in this check? if yes then what check should violate this example I think we don't have another check that enforces wrapping

**Comment by rnveach** [comment]
I assume when is just a keyword then?

This seems like it might have been our fault to add method reference if it isn't a operator (was added in #3472), but this would need to be a breaking change to fix this. Let's see what @nrmancuso and @romani  thinks.

> what check should violate this example I think we don't have another check that enforces wrapping

This sounds like it would be a new check, but this would add more questions for me. Do we have something that says `when` should be wrapped like this issue specifies?

**Comment by nrmancuso** [comment]
>This seems like it might have been our fault to add method reference if it isn't a operator 

Yep, method reference should be in https://checkstyle.sourceforge.io/checks/whitespace/separatorwrap.html#SeparatorWrap, we can open an issue for this and discuss there.

As far as this issue goes, I am OK to let this simmer a bit (end of summer or beyond) and see what practices emerge as more users exercise this syntax. I am going to update the issue name to make it easier to find by Github search. 

That said, I tend to think that we might just want to make a new check for this like 'KeywordWrapping` or something.

**Comment by nrmancuso** [comment]
My thoughts on this one are as follows: If we had a check that checked `instanceof` for wrapping, we would likely update that check (I couldn't find one).

But, this seems like a pretty niche requirement, and fairly pedantic. I am going to remove this from our project board, and wait until we see some community demand for something like this. Maybe `KeywordWrappingCheck`.
